PLFM-4266

kimyen opened the pull request:
Extend ACTAccessRequirement object

kimyen committed:
Extend ACTAccessRequirement object

kimyen committed:
return ACTAccessRequirement with default values

brucehoff commented on lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/ACTAccessRequirement.json:
```diff
@@ -15,6 +15,38 @@
+		},
+		"isDUCRequired": {
+			"type":"boolean",
+			"description":  "If true, then accessor needs to fill, sign, and submit a DUC to gain access to the data."
```
Data Use Certificate (DUC)

brucehoff commented on lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/ACTAccessRequirement.json:
```diff
@@ -15,6 +15,38 @@
+		},
+		"isIRBRequired": {
+			"type":"boolean",
+			"description":  "If true, then accessor needs to submit an IRB to gain access to the data."
```
Perhaps "IRB" should be "IRB approval" or "IRB approval document".

brucehoff commented on services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AccessRequirementManagerImpl.java:
```diff
@@ -232,9 +226,45 @@ public void deleteAccessRequirement(UserInfo userInfo,
 				authorizationManager.canAccess(userInfo, accessRequirementId, ObjectType.ACCESS_REQUIREMENT, accessType));
 	}
+
+	static AccessRequirement setDefaultValues(AccessRequirement ar) {
```
My suggestion is to put this logic in the migration translator for the AR DBO. Also this manager can ensure that these fields are set upon creation or update.  This could be called 'setACTAccessRequirementDefaultValues()'.

kimyen committed:
Merge remote-tracking branch 'upstream/develop' into PLFM-4266

kimyen committed:
update doc

kimyen committed:
refactor

instead of changing the returning object, migrate old data to have
default values and set default values for new ACTAccessRequirement.

brucehoff commented:
Great job!
